infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/Labels.java

```diff
@@ -170,6 +189,19 @@ public RuntimeIdentity runtimeId() {
       }
       return servers;
     }
+
+    /** Retrieves machine attributes from docker labels and returns them. */
+    public Map<String, String> machineAttributes() {
+      final Map<String, String> attributes = new HashMap<>();
+      for (Entry<String, String> entry : labels.entrySet()) {
+        Matcher matcher = MACHINE_ATTRIBUTE_PATTERN.matcher(entry.getKey());
+        if (matcher.matches()) {
+          String attribute = matcher.group("attribute");
```

Comment:
I think it is better to serialize/deserialize map instead of using single attribute.